File: jevents-api/app/models/payment.rb
class Payment < ApplicationRecord
  belongs_to :order

  enum :method, {
    upi: 0,
    credit_card: 1,
    debit_card: 2,
    netbanking: 3,
    coupon: 4
  }

  enum :status, {
    initiated: 0,
    success: 1,
    failed: 2,
    refunded: 3
  }

  # validates :amount, presence: true, numericality: { greater_than: 0 }
  # validates :transaction_id, length: { maximum: 100 }, allow_nil: true
end

File: jevents-ui/src/types/payment.ts
export interface Payment {
  id: number;
  order_id: number;
  method: "upi" | "credit_card" | "debit_card" | "netbanking" | "coupon";
  transaction_id: string;
  amount: number;
  status: "initiated" | "success" | "failed" | "refunded";
  created_at: string;
  updated_at: string;
}

export interface CreatePaymentRequest {
  order_id: number;
  method: Payment["method"];
  transaction_id: string;
  amount: number;
  status: Payment["status"];
}

export interface UpdatePaymentRequest {
  method?: Payment["method"];
  status?: Payment["status"];
}

export interface PaymentResponse {
  payment: Payment;
}
